luavm/src/lib/lua_values/mod.rs

```diff
@@ -51,9 +58,11 @@ impl LuaVal {
                 // https://www.lua.org/manual/5.3/manual.html#3.4.3
                 // The behaviour of `as f64` is the same as the conversion
                 // from int to float described in the manual.
-                LuaValKind::INT => Ok(*(i64_ptr(self.encoded_ptr)) as f64),
-                LuaValKind::FLOAT => Ok(*(f64_ptr(self.encoded_ptr))),
-                LuaValKind::STR => (*(str_ptr(self.encoded_ptr)))
+                LuaValKind::INT => Ok(((self.val >> 3) as i64) as f64),
+                // make sure to remove the encoded tag
```

Comment:
What does this refer to?

Reply:
It refers to the float case, is it misleading?

Reply:
Yes, better to put the expression in curly brackets and put the comment inside it IMHO.